fastapi/pools/event_loop.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2025 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
+
+import asyncio
+import queue
+import threading
+from contextlib import contextmanager
+from typing import Generator
+
+
+class EventLoopPool:
+    def __init__(self, max_size=0):
+        self.max_size = max_size
+        self.pool = queue.Queue[tuple[asyncio.AbstractEventLoop, threading.Thread]](
+            maxsize=max_size
+        )
+
+        # Pre-create event loop threads up to the pool size
+        for _ in range(max_size):
+            loop = asyncio.new_event_loop()
+            thread = threading.Thread(target=loop.run_forever, daemon=True)
+            thread.start()
+            self.pool.put((loop, thread))
+
+    def __get_event_loop_and_thread(
+        self,
+    ) -> tuple[asyncio.AbstractEventLoop, threading.Thread]:
+        """
+        Get an event loop from the pool. If no event loop is available, create a new one.
+        """
+        try:
+            return self.pool.get_nowait()
```

Comment:
If max_size is set, should this not be `get()` ? Otherwise if the pool is empty, it will create a new thread and possibly block when putting threads back in the pool when it is full again?

Reply:
Which is a theoretical concern since EventPool does not seem to ever be initialized with a max_size. So perhaps remove support for max_size for the sake of simplicity, knowing that the native Odoo mechanism will limit the number of concurrent threads?

Reply:
I'll remove max size. It's useless and could conflict with the native Odoo mechanism.